crates/sui-graphql-rpc/src/consistency.rs

```diff
@@ -166,14 +167,16 @@ pub(crate) fn build_objects_query(
                 newer
             );
             history_objs = filter!(history_objs, "newer.object_version IS NULL");
+            history_objs = filter!(history_objs, "object_status = 0");
             history_objs
         }
         View::Historical => {
             // The cursor pagination logic refers to the table with the `candidates` alias
-            query!(
+            let query = query!(
                 "SELECT candidates.* FROM ({}) candidates",
                 history_objs_inner
-            )
+            );
+            filter!(query, "object_status = 0")
```

Comment:
is this doing history object lookup, iiuc we will start pruning objects_history to 2 partitions, should we start reading this from `full_objects_history` instead.

Reply:
This is fine and desired because we only want to find yet newer versions of any objects that matched the filtering conditions (candidates)